File: bench/elixir/pipeline.exs
n = String.to_integer(System.get_env("BENCH_N") || "200000")

sum =
  0..(n - 1)
  |> Stream.filter(fn i -> rem(i, 3) == 0 or rem(i, 5) == 0 end)
  |> Stream.map(fn i -> i * i end)
  |> Enum.reduce(0, &+/2)

IO.puts(sum)

File: bench/clojure/pipeline.clj
;; filter -> map -> reduce, fused via a transducer (the Clojure streaming idiom).
(let [n (long (Long/parseLong (or (System/getenv "BENCH_N") "100000")))]
  (println (transduce (comp (filter (fn [i] (or (zero? (rem i 3)) (zero? (rem i 5)))))
                            (map (fn [i] (* i i))))
                      + 0 (range n))))
